src/myspace/AdditionalInfo.jsx

```diff
@@ -1,21 +1,149 @@
-import React from "react";
+import React, { useState, useEffect, useContext } from "react";
+import { BASE_API } from "../config";
+import { AuthContext } from "../AuthContext";
+import "./MySpace.css";
+
 
 export default function AdditionalInfo() {
-    return (
-        <div className="container-fluid" id="profile">
-          <div className="top">
-            <h1>
-              This is the Private page for Member's Additional Info page
-            </h1>
-          </div>
-          <div className="middle">
-            <h2>The site is currently under construction...</h2>
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [additionalInfo, setAdditionalInfo] = useState({});
+  const { access_token, user } = useContext(AuthContext);
+
+  const requestAdditionalInfo = {
+    method: "GET",
+    headers: {
+      "Authorization": `Bearer ${access_token}`,
+      "Accept": "application/json",
+      "Content-Type": "application/json",
+    },
+  };
+
+
+  let data = "";
+  useEffect(() => {
+    fetch(`${BASE_API}/user/additional_info`, requestAdditionalInfo)
+      .then(async response => {
+        data = await response.json();
+        if (response.ok) {
+          return setAdditionalInfo(data);
+        } else {
+          throw new Error(data.message);
+        }
+      })
+      .catch(error => {
+        setErrorMessage(data.message);
```

Comment:
if you throw an error, and send the data.message, in your catch, you will get access to that data.message. So you should be able to say setErrorMessage(error) 

while throwing simply throw new Error(data)

once you do that in catch you can say .catch(error) {set.... error.message}

Please note: whatever you send inside new Error will come to catch

Have you tried this out locally, when you run the app? You might want to check with a 500 scenario, and a 400 scenario both

Reply:
Yep @meenakshi-dhanani . I tried it (with the existing code). The error is shown when user didn't retrieved their details before viewing additional info page.

![ezgif com-video-to-gif (3)](https://user-images.githubusercontent.com/29667122/88244696-eb8ad280-ccd7-11ea-99d0-856da64b6584.gif)

Note that there's a bug in this application. Logout on frontend currently only removes the cookies on frontend to allow toggling navbar with isAuth status. However, no logout being implemented on backend, thus logging out from frontend does not remove cookies of user_id in backend. At this stage, to remove this cookies on backend, the bit backend seerver needs to be terminated completely and ree-run again for frontend to login. 

See the behaviour below. User was logged-in, then logged-out, then login again. Expected behaviour is for user to be warned that they haven't retrieved their personal details. But instead, additional info is shown as if they had done so (retrieved their details)

![ezgif com-video-to-gif (4)](https://user-images.githubusercontent.com/29667122/88244969-c2b70d00-ccd8-11ea-89f6-0c0d881b875b.gif)

Reply:
Update @meenakshi-dhanani . I've just modified throw as per your suggestion. It still works the same as gif above